### ISSUE TITLE
feat: Viper should only check for Northstar updates, after checking for Viper updates

### ISSUE THREAD

**Issue by GeckoEidechse** [0neGal/viper#72]
**What feature would you like added?**
Viper should prioritise checking for its own updates over Northstar updates, checking for Northstar updates only after the Viper update check has been completed.

**Why should this feature be added?**
The idea is that if anyone uses an older version of Viper that might have issues with a newer version of Northstar, said version of Viper won't be able to download the Northstar update until the user either updated Viper or dismissed the update dialogue.
Such a situation hasn't occurred yet nor might it ever occur but the idea is should there ever be a moment where Viper has issues with a new release of Northstar, an update for Viper can be pushed that resolves the problem and Viper users will get the update for Viper before updating Northstar.
Conversely without this, say a new Northstar version causes a crash for Viper when trying to download it. User's will restart their Viper client and it tries to auto-download Northstar again, causing it yet to crash. Viper won't auto-update due to Viper updates not being prioritised. Users will have to manually update Viper by downloading the release from Github to resolve the hypothetical crash-loop scenario.

**Additional Info**
Overall this is a pretty low priority feature until it's too late already :P

**Comment by Alystrasz** [comment]
I see pretty clearly how to prevent checking for Northstar updates while we're checking for Viper updates, but I don't exactly get why would Viper be unable to download newest Northstar version 😅

**Comment by 0neGal** [comment]
> but I don't exactly get why would Viper be unable to download newest Northstar version

Say Northstar changes repo, file structure or anything alike, the old versions will no longer be able to install Northstar, and a version which prioritizes Viper updates would then stop problems from occurring, that is if we implement the new format.

It's more of a contingency plan. Since you've a good idea of how electron-builder works and you also seem to know how to implement this I'll assign you to the issue, unless you would rather not be assigned :)

**Comment by GeckoEidechse** [comment]
> Say Northstar changes repo, file structure or anything alike, the old versions will no longer be able to install Northstar, and a version which prioritizes Viper updates would then stop problems from occurring, that is if we implement the new format.

This. If there's a change with Northstar file structure etc we would of course, give everyone a heads-up. ^^
It's kinda more for cases when we do minor changes that hit some edge case for Viper causing it to get stuck in a crash loop.

Basically:
> It's more of a contingency plan.

**Comment by Alystrasz** [comment]
I've done it by basically launching N* updates check when we detect no Viper update is available: https://github.com/Alystrasz/viper/blob/fix/updates-order/src/utils.js#L310
